packages/@react-aria/dnd/src/DragManager.ts

```diff
@@ -188,7 +196,23 @@ class DragSession {
     );
     this.updateValidDropTargets();
 
-    announce(this.formatMessage(MESSAGES[getDragModality()]));
+    let dragModality = getDragModality();
+    announce(this.formatMessage(MESSAGES[dragModality]));
+
+    if (dragModality === 'virtual') {
+      this.genericAncestor =
+        Array.from(document.body.querySelectorAll(':not([aria-hidden]):not([role])'))
+        .find(
+          element => (
+            element.contains(this.dragTarget.element) &&
+            this.validDropTargets.every(target => element.contains(target.element))
+          )
+        );
+      if (this.genericAncestor) {
+        this.genericAncestor.setAttribute('role', 'application');
+        this.genericAncestor.setAttribute('aria-label', this.formatMessage('dragApplicationLabel'));
```

Comment:
i know we're looking for elements without a role, so it's fine to set and wipe that out, but can an element without a role have an aria-label already? should we restore it afterwards if so?

Reply:
I suppose, if the container is a named HTML5 landmark, it could have an `aria-label`. If so, I can cache and restore it.